examples/tuh_eeg_corpus.py

```diff
@@ -0,0 +1,258 @@
+    raw.rename_channels(mapping[reference])
+
+
+def custom_crop(raw, tmin=0.0, tmax=None, include_tmax=True):
+    # crop recordings to tmin – tmax. can be incomplete if recording
+    # has lower duration than tmax
+    # by default mne fails if tmax is bigger than duration
+    tmax = min((raw.n_times - 1) / raw.info['sfreq'], tmax)
+    raw.crop(tmin=tmin, tmax=tmax, include_tmax=include_tmax)
+
+
+tmin = 1*60
+tmax = 6*60
+sfreq = 100
+
+preprocessors = [
+    MNEPreproc(custom_crop, tmin=tmin, tmax=tmax, include_tmax=False),
+    MNEPreproc('set_eeg_reference', ref_channels='average', ch_type='eeg'),
+    MNEPreproc(custom_rename_channels, mapping=ch_mapping),
+    MNEPreproc("pick_channels", ch_names=short_ch_names, ordered=True),
+    NumpyPreproc(lambda x: x * 1e6),
+    MNEPreproc("resample", sfreq=sfreq),
+]
+
+
+###############################################################################
+# The preprocessing loop works as follows. For every recording, we apply the
+# preprocessors as defined above. Then, we update the description of the rec,
+# since we have altered the duration, the reference, and the sampling frequency.
+# Afterwards, we split the continuous signals into compute windows. We store
+# each recording to a unique subdirectory that is named corresponding to the
+# rec id. To save memory, after windowing and storing, we delete the raw
+# dataset and the windows dataset, respectively.
+window_size_samples = 1000
+window_stride_samples = 1000
+create_compute_windows = True
+
+out_i = 0
+errors = []
+OUT_PATH = './tuh_sample/'
+tuh_splits = tuh.split([[i] for i in range(len(tuh.datasets))])
+for rec_i, tuh_subset in tuh_splits.items():
+    # implement preprocess for BaseDatasets? Would remove necessity
+    # to split above
+    preprocess(tuh_subset, preprocessors)
+
+    # update description of the recording(s)
+    tuh_subset.description.sfreq = len(tuh_subset.datasets) * [sfreq]
+    tuh_subset.description.reference = len(tuh_subset.datasets) * ['ar']
+    tuh_subset.description.n_samples = [len(d) for d in tuh_subset.datasets]
+
+    if create_compute_windows:
+        # generate compute windows here and store them to disk
+        tuh_windows = create_fixed_length_windows(
+            tuh_subset,
+            start_offset_samples=0,
+            stop_offset_samples=None,
+            window_size_samples=window_size_samples,
+            window_stride_samples=window_stride_samples,
+            drop_last_window=False
+        )
+        # save memory by deleting raw recording
+        del tuh_subset
+        # store the number of windows required for loading later on
+        tuh_windows.description["n_windows"] = [len(d) for d in
+                                                tuh_windows.datasets]
+
+        # some recordings fail with
+        # info["meas_date"] seconds must be between "(-2147483648, 0)" and "(2147483647, 0)", got "-2209161600"  # noqa
+        # during save, so catch it
+        try:
```

Comment:
Saving of recordings sometimes failed due to an invalid measurement date. I was able to detect these invalid dates using `mne.io.maes_info._check_dates`. However, this does not return `True` or `False` but raises a `RuntimeError` instead. Is there a better way to handle this @agramfort ?